device-management-service/src/service/mod.rs

```diff
@@ -460,6 +469,26 @@ where
         Ok(app.map(Into::into))
     }
 
+    async fn list_apps(
+        &self,
+        identity: &dyn Identity,
+    ) -> Result<Option<Vec<Application>>, Self::Error> {
+        let c = self.pool.get().await?;
+
+        let apps = PostgresApplicationAccessor::new(&c)
+            .list(Lock::None)
+            .await?;
+
+        let mut apps_return = Vec::new();
+        for app in apps {
+            if ensure(&app, identity).is_ok() {
+                apps_return.push(Into::into(app));
+            }
+        }
```

Comment:
This can probably be done in a nicer way using something like : 
```
        apps.iter().map(|app| {
            if ensure(app, identity).is_ok() {
                Application::from(app)
            }
        } ).collect();
```
but if ensure returns an error the item should be removed from the Vector so i am not sure ow to do it

Reply:
How about `apps.iter().filter(|app| ensure(app, identity).is_ok()).map(|app| Application::from(app)).collect()`